test/setup_bundler_test.rb

```diff
@@ -91,14 +84,38 @@ def test_loading_custom_bundle_dependencies_is_lazy
   end
 
   def test_does_not_copy_gemfile_lock_when_not_modified
-    Object.any_instance.expects(:system).with(bundle_env(".ruby-lsp/Gemfile"), "bundle install 1>&2")
-    Bundler::LockfileParser.any_instance.expects(:dependencies).returns({}).twice
-    FileUtils.mkdir(".ruby-lsp")
-    FileUtils.cp("Gemfile.lock", ".ruby-lsp/Gemfile.lock")
+    Dir.mktmpdir do |dir|
+      Dir.chdir(dir) do
+        File.write(File.join(dir, "Gemfile"), <<~GEMFILE)
+          source "https://rubygems.org"
+          gem "rdoc"
+        GEMFILE
 
-    run_script
-  ensure
-    FileUtils.rm_r(".ruby-lsp")
+        capture_subprocess_io do
+          Bundler.with_unbundled_env do
+            # Run bundle install to generate the lockfile
+            system("bundle install")
+
+            # Run the script once to generate a custom bundle
+            run_script
+          end
+        end
+
```

Comment:
Can we update the `mtime` of the lockfile in between these steps to ensure that it doesn't get copied when all that changes is the last modified time and not the digest? We should be able to do it with `FileUtils.touch("Gemfile.lock", mtime: Time.now + 10 * 60)`